blog/_posts/2017-09-12-Issue-253.md

```diff
@@ -0,0 +1,134 @@
+
+1. create Feedback field to Food Pickup ( [Issue #235](https://github.com/yunity/foodsaving-backend/issues/253) )
+2. make trusted users/group admins    
+3. create statistics for pickups that were done by users in kilograms   
+4. build a quiz framework   
+
+Optional: Android App, shared food & other things
+
+Yay!
+
+<br>
+
+## The Reality
+We found out that we had no clue how difficult such an Issue can be! I mean like ABSOLUTLEY no clue. Coming from 'Codecademy', 'Learn Python the Hard Way' and some small pet projects we thought we would solve the first Issue within 1 week.    
+
+We needed 9 weeks until our first (noteable) Pull request was accepted. So, how was that possible? 
+
+<br>
+
+## The First Issue
+
+Our first task was to create Feedback field to Food Pickups. We spent a lot of our time learning about: Python, Django Framework, Django REST Framework, Swagger, how to troubleshoot our Git merge conflicts and how to solve our Docker complaints.
+
+
+#### We Have Learnt:    
+1. There are no shortcuts - we have to do Django tutorial and Django REST tutorial first  
+2. Working both on the same branch at the same time will not speed up our progress! We are going to have many merge conflicts instead and we have to solve them all. This takes many, many hours :)  
+3. We know what is server, browser, database, models, migrations, HTTP requests, querysets, filters and all these fancy words around we did not know how they are connected before.
+4. And finally: This Issue is not as small as it looks like!
+
+If you want to read more about this, go to the Foodsaving blog and read the [Fairy Tale Version](https://blog.foodsaving.world/2017/09/01/railsgirls-fairy-tale.html) of that story.
+
+<a href="https://blog.foodsaving.world/2017/09/01/railsgirls-fairy-tale.html"> <img src="/img/blog/2017/Issue253.jpg" width="300"></a><br>
+_Graphic by Ines D. Guett_
+
+
+#### We Have Built (Status: 6. September 2017):   
+- a new API endpoint Feedback with HTTP methods GET, POST  
+- integration API tests for Feedback that ensure that the user gets the correct GET/POST output  
+- validation for field ```about``` and set the field ```given_by``` as read-only
+- unit model tests for model Feedback
+
+<br>
+
+Our hardest part on our Issue was to fix this test:
+```python
+    def test_list_feedback_works_as_collector(self):    
+        """
+        Collector is allowed to see list of feedback.
+        """
+        self.client.force_login(user=self.collector)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertEqual(len(response.data), 2)
+```
+This test should return two instances of our test object and check that there are exactly 2 instances in the test database. There is also a difference between a test database and live database. We discussed this many times until we could fully understand :)
```

Comment:
This test should return two instances of our test object and check that there are exactly two instances in the test database. There is also a difference between a test database and a live database. We discussed this many times until we would fully understand :)